core/src/main/java/net/opentsdb/query/processor/bucketquantile/BucketQuantileConfig.java

```diff
@@ -0,0 +1,618 @@
+        .setCumulativeBuckets(cumulative_buckets)
+        .setCounterBuckets(counter_buckets)
+        .setNanThreshold(nan_threshold)
+        .setMissingMetricThreshold(missing_metric_threshold);
+    super.toBuilder(builder);
+    return builder;
+  }
+  
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    final BucketQuantileConfig config = (BucketQuantileConfig) o;
+    return Objects.equal(as, config.getAs()) &&
+        Objects.equal(bucket_regex, config.bucket_regex) &&
+        Objects.equal(overflow, config.overflow) &&
+        Objects.equal(overflow_max, config.overflow_max) &&
+        Objects.equal(underflow, config.underflow) &&
+        Objects.equal(underflow_min, config.underflow_min) &&
+        Objects.equal(output_of_bucket, config.output_of_bucket) &&
+        Objects.equal(histograms, config.getHistograms()) &&
+        Objects.equal(quantiles, config.quantiles) &&
+        Objects.equal(cumulative_buckets, config.cumulative_buckets) &&
+        Objects.equal(counter_buckets, config.counter_buckets) &&
+        Objects.equal(nan_threshold, config.nan_threshold) &&
+        Objects.equal(missing_metric_threshold, config.missing_metric_threshold) &&
+        Objects.equal(infectious_nan, config.getInfectiousNan()) &&
+        Objects.equal(interpolator_configs, config.interpolator_configs) &&
+        Objects.equal(id, config.getId());
+  }
+  
+  @Override
+  public int hashCode() {
+    return buildHashCode().asInt();
+  }
+  
+  /** @return A HashCode object for deterministic, non-secure hashing */
+  public HashCode buildHashCode() {
+    final List<HashCode> hashes =
+        Lists.newArrayListWithCapacity(3);
+    hashes.add(super.buildHashCode());
+        
+    final Hasher hasher = Const.HASH_FUNCTION().newHasher();
+    hasher.putString(bucket_regex, Const.UTF8_CHARSET)
+          .putDouble(overflow_max)
+          .putString(overflow == null ? "" : overflow, Const.UTF8_CHARSET)
+          .putDouble(underflow_min)
+          .putString(underflow == null ? "" : underflow, Const.UTF8_CHARSET)
+          .putInt(output_of_bucket.ordinal())
+          .putString(as, Const.UTF8_CHARSET)
+          .putBoolean(cumulative_buckets)
+          .putBoolean(counter_buckets)
+          .putDouble(nan_threshold)
+          .putDouble(missing_metric_threshold);
+    for (int i = 0; i < histograms.size(); i++) {
+      hasher.putString(histograms.get(i), Const.UTF8_CHARSET);
+    }
+    for (int i = 0; i < quantiles.size(); i++) {
+      hasher.putDouble(quantiles.get(i));
+    }
+    hasher.putString(as, Const.UTF8_CHARSET)
+          .putBoolean(infectious_nan);
+    hashes.add(hasher.hash());
+    return Hashing.combineOrdered(hashes);
+  }
+
+  @Override
+  public int compareTo(final BucketQuantileConfig o) {
+    // TODO Auto-generated method stub
```

Comment:
Don't need this?

Reply:
No not right now, I haven't been using it and need to yank it.